lapack/gonum/dorgr2.go

```diff
@@ -54,16 +49,16 @@ func (impl Implementation) Dorgr2(m, n, k int, a []float64, lda int, tau, work [
 		panic(badLenTau)
 	case len(a) < (m-1)*lda+n:
 		panic(shortA)
+	case len(work) < m:
+		panic(shortWork)
 	}
 
-	if k < m {
-		// Initialise rows 0:m-k to rows of the unit matrix.
-		for l := 0; l < m-k; l++ {
-			for j := 0; j < n; j++ {
-				a[l*lda+j] = 0
-			}
-			a[l*lda+n-m+l] = 1
+	// Initialise rows 0:m-k to rows of the unit matrix.
+	for l := 0; l < m-k; l++ {
+		for j := 0; j < n; j++ {
+			a[l*lda+j] = 0
 		}
```

Comment:
Is it worth helping the compiler optimise this?
```
row := a[l*lda:l*lda+n]
for j := range row {
    row[j] = 0
}
```

Reply:
Good idea, done.